caddyhttp/proxy/reverseproxy.go

```diff
@@ -189,7 +189,7 @@ func (rp *ReverseProxy) ServeHTTP(rw http.ResponseWriter, outreq *http.Request,
 		res.Body.Close()
 		hj, ok := rw.(http.Hijacker)
 		if !ok {
-			return nil
+			panic("not a hijacker")
```

Comment:
Well, I think it should panic here because it's a programmer's error instead of a common user error and this panic should be captured by [here](https://github.com/mholt/caddy/blob/master/caddyhttp/httpserver/server.go#L205).

Also if it returns specific error here, proxy's `ServeHTTP` will retry some times.
Although we could make a exception for this specific error, but I think it adds some verbosity.